rosidl_generator_cpp/test/test_msg_builder.cpp

```diff
@@ -0,0 +1,205 @@
+    .uint32_values_default({{120000UL, 60000UL, 120000UL}})
+    .int64_values_default({{-8000000LL, -4000000LL, -8000000LL}})
+    .uint64_values_default({{10000000ULL, 5000000ULL, 10000000ULL}})
+    .string_values_default({{"test test", "test", "test test"}})
+    .alignment_check(0l);
+
+  rosidl_generator_cpp::msg::BoundedSequences bounded_sequences;
+  rosidl_generator_cpp::msg::UnboundedSequences unbounded_sequences;
+
+  rosidl_generator_cpp::msg::MultiNested multi_nested =
+    rosidl_generator_cpp::build<rosidl_generator_cpp::msg::MultiNested>()
+    .array_of_arrays({{arrays, arrays, arrays}})
+    .array_of_bounded_sequences(
+    {{
+      bounded_sequences,
+      bounded_sequences,
+      bounded_sequences
+    }})
+    .array_of_unbounded_sequences(
+    {{
+      unbounded_sequences,
+      unbounded_sequences,
+      unbounded_sequences
+    }})
+    .bounded_sequence_of_arrays({arrays, arrays})
+    .bounded_sequence_of_bounded_sequences(
+  {
+    bounded_sequences,
+    bounded_sequences
+  })
+    .bounded_sequence_of_unbounded_sequences(
+  {
+    unbounded_sequences,
+    unbounded_sequences
+  })
+    .unbounded_sequence_of_arrays({arrays})
+    .unbounded_sequence_of_bounded_sequences({bounded_sequences})
+    .unbounded_sequence_of_unbounded_sequences({unbounded_sequences});
+
+  for (const auto & a : multi_nested.array_of_arrays) {
+    ASSERT_EQ(arrays, a);
+  }
+
+  for (const auto & b : multi_nested.array_of_bounded_sequences) {
+    ASSERT_EQ(bounded_sequences, b);
+  }
+
+  for (const auto & u : multi_nested.array_of_unbounded_sequences) {
+    ASSERT_EQ(unbounded_sequences, u);
+  }
+
+  ASSERT_EQ(2u, multi_nested.bounded_sequence_of_arrays.size());
+  for (const auto & a : multi_nested.bounded_sequence_of_arrays) {
+    ASSERT_EQ(arrays, a);
+  }
+
+  ASSERT_EQ(2u, multi_nested.bounded_sequence_of_bounded_sequences.size());
+  for (const auto & b : multi_nested.bounded_sequence_of_bounded_sequences) {
+    ASSERT_EQ(bounded_sequences, b);
+  }
+
+  ASSERT_EQ(2u, multi_nested.bounded_sequence_of_unbounded_sequences.size());
+  for (const auto & u : multi_nested.bounded_sequence_of_unbounded_sequences) {
+    ASSERT_EQ(unbounded_sequences, u);
+  }
+
+  ASSERT_EQ(1u, multi_nested.unbounded_sequence_of_arrays.size());
+  for (const auto & a : multi_nested.unbounded_sequence_of_arrays) {
+    ASSERT_EQ(arrays, a);
+  }
+
+  ASSERT_EQ(1u, multi_nested.unbounded_sequence_of_bounded_sequences.size());
+  for (const auto & b : multi_nested.unbounded_sequence_of_bounded_sequences) {
+    ASSERT_EQ(bounded_sequences, b);
+  }
+
+  ASSERT_EQ(1u, multi_nested.unbounded_sequence_of_unbounded_sequences.size());
+  for (const auto & u : multi_nested.unbounded_sequence_of_unbounded_sequences) {
+    ASSERT_EQ(unbounded_sequences, u);
+  }
+
+  rosidl_generator_cpp::msg::Empty empty =
+    rosidl_generator_cpp::build<rosidl_generator_cpp::msg::Empty>()
+    .structure_needs_at_least_one_member(5);
```

Comment:
@dirk-thomas is this still necessary? I thought we had dropped that restriction.

Reply:
> I thought we had dropped that restriction.

Where do you think that happened? This PR doesn't does the special member in any way.